### PR TITLE
add the redundant/degraded option

### PR DESCRIPTION
**Pull request by caberos**
https://github.com/softlayer/softlayer-python/issues/1257

add the redundant/degraded option

### CODE REVIEW

File: SoftLayer/CLI/hardware/edit.py [hunk 1]
```diff
@@ -51,7 +53,17 @@ def cli(env, identifier, domain, userfile, tag, hostname, userdata, public_speed
         raise exceptions.CLIAbort("Failed to update hardware")
 
     if public_speed is not None:
-        mgr.change_port_speed(hw_id, True, int(public_speed))
+        if rebundant:
```
Comment (allmightyspiff) [inline SoftLayer/CLI/hardware/edit.py:56]: misselling, `redundant`

File: SoftLayer/CLI/hardware/edit.py [hunk 2]
```diff
@@ -22,8 +22,10 @@
               help="Public port speed. -1 is best speed available")
 @click.option('--private-speed', default=None, type=click.Choice(['0', '10', '100', '1000', '10000', '-1']),
               help="Private port speed. -1 is best speed available")
+@click.option('--rebundant', is_flag=True, default=False, help="The desired state of redundancy for the interface(s)")
```
Comment (allmightyspiff) [inline SoftLayer/CLI/hardware/edit.py:25]: make sure to change all `rebundant` to `redundant`

File: SoftLayer/managers/hardware.py [hunk 3]
```diff
@@ -319,11 +319,11 @@ def change_port_speed(self, hardware_id, public, speed):
         if public:
             return self.client.call('Hardware_Server',
                                     'setPublicNetworkInterfaceSpeed',
-                                    speed, id=hardware_id)
+                                    [rebundant, speed], id=hardware_id)
```
Comment (allmightyspiff) [inline SoftLayer/managers/hardware.py:322]: redundant should be the second parameter, not the first. https://sldn.softlayer.com/reference/services/SoftLayer_Hardware_Server/setPublicNetworkInterfaceSpeed/

File: SoftLayer/managers/hardware.py [hunk 4]
```diff
@@ -319,11 +319,11 @@ def change_port_speed(self, hardware_id, public, speed):
...
         else:
             return self.client.call('Hardware_Server',
                                     'setPrivateNetworkInterfaceSpeed',
-                                    speed, id=hardware_id)
+                                    [rebundant, speed], id=hardware_id)
```
Comment (allmightyspiff) [inline SoftLayer/managers/hardware.py:326]: redundant should be the second parameter, not the first. https://sldn.softlayer.com/reference/services/SoftLayer_Hardware_Server/setPublicNetworkInterfaceSpeed/

File: SoftLayer/managers/hardware.py [hunk 5]
```diff
@@ -297,7 +297,7 @@ def rescue(self, hardware_id):
         """
         return self.hardware.bootToRescueLayer(id=hardware_id)
 
-    def change_port_speed(self, hardware_id, public, speed):
+    def change_port_speed(self, hardware_id, public, speed, rebundant):
```
Comment (allmightyspiff) [inline SoftLayer/managers/hardware.py:300]: redundant should default to None if not supplied.
`def change_port_speed(self, hardware_id, public, speed, redundant=None):`

**Comment by allmightyspiff** [comment]
Pull in the new changes from master, and the tests should pass.

**Comment by dkulchinsky** [comment]
@caberos @allmightyspiff any chance to cut a release with this new feature? we're really looking forward to it 🙏

**Comment by allmightyspiff** [comment]
Sure, I can add this to a release on Monday

**Pull request merged**